frontend/lib/src/components/shared/Dropdown/Selectbox.tsx

```diff
@@ -88,222 +77,205 @@ export function fuzzyFilterSelectOptions(
   ).reverse()
 }
 
-export class Selectbox extends React.PureComponent<Props, State> {
-  public state: State = {
-    isEmpty: false,
-    value: this.props.value,
-  }
+const Selectbox: React.FC<Props> = ({
+  disabled,
+  width,
+  value: propValue,
+  onChange,
+  options: propOptions,
+  label,
+  labelVisibility,
+  help,
+  placeholder,
+  clearable,
+}) => {
+  const theme: EmotionTheme = useTheme()
+  const [isEmpty, setIsEmpty] = useState(false)
```

Comment:
**suggestion:** This looks like it can be derived from the `value` state, so this should become a `useMemo` instead

Reply:
I was actually able to fully remove the isEmpty state. It issue why it was added is not anymore reproducable with the baseweb selectbox:

https://github.com/streamlit/streamlit/pull/3296
https://github.com/streamlit/streamlit/issues/3191

Everything seems to work fine without the `isEmpty` state. Maybe it was fixed by baseweb or somehow via the conversion to functional.